Tests/Functional/FailingTestSuite/main.swift

```diff
@@ -55,7 +55,9 @@ class FailingTestCase: XCTestCase {
     }
 }
 
+extension PassingTestCase: XCTestCaseType {}
+extension FailingTestCase: XCTestCaseType {}
```

Comment:
In the grand scheme of things, I feel this is fairly painless for end-users. :+1: 

If they're already writing cross-platform tests (like the swift-corelibs-foundation and swift-package-manager projects are), then they have a conditionally-compiled file named that's used to call `XCTMain()`. This pull request simply requires those projects to add extensions like these in that file, one for each of their test cases.

Once we can use runtime reflection to automatically discover tests, we can simplify this quite a bit--we won't need to jump through hoops to ensure we don't `override` `allTests`. In the meantime, this seems like the best way to proceed.